### ISSUE TITLE
Incorrect gradfun for torch.RepeatTensor

### ISSUE THREAD

**Issue by OptimusLime** [twitter/torch-autograd#91]
I cannot submit a pull request right now, but there is an issue using torch.RepeatTensor with non-double tensors. 

Within the gradient function for torch.RepeatTensor in [gradfuns.lua](https://github.com/twitter/torch-autograd/blob/master/src/gradfuns.lua#L503), torch.cat is called but this is incorrect for non-double tensors. 

The correct module should call util.cat. The new module would be: 

```lua
 module.gradient("repeatTensor", {
      function(g, ans, x, ...)
         local Dg = torch.nDimension(g)
         local Dx = torch.nDimension(x)
         for i=Dx,1,-1 do
            local D = torch.nDimension(g)
            local c = util.cat(torch.split(g,torch.size(x,i), Dg-Dx+i), D+1)
            g = torch.squeeze(torch.sum(c,D+1))
         end
         for i=1,Dg-Dx do
            g = torch.squeeze(torch.sum(g,1))
         end
         return g
      end,
      function(g, ans, ...) return nil end,
      function(g, ans, ...) return nil end,
      function(g, ans, ...) return nil end,
      function(g, ans, ...) return nil end,
      function(g, ans, ...) return nil end, -- five dimensions should be enough
   })
```

To expose the error, you can augment the RepeatTensor test to include a cuda tensor as input: 

```lua 
   RepeatTensor = function()
      local function f2to2(params)
         local y = torch.repeatTensor(params.x, 2, 2)*3
         return torch.sum(y)
      end
      tester:assert(gradcheck(f2to2, {x=torch.randn(3,3)}), "Incorrect gradient")

      local function f3to3(params)
         local y = torch.repeatTensor(params.x, 2, 2, 2)*3
         return torch.sum(y)
      end
      tester:assert(gradcheck(f3to3, {x=torch.randn(3,3,3)}), "Incorrect gradient")

      local function f2to3(params)
         local y = torch.repeatTensor(params.x, 2, 2, 2)*3
         return torch.sum(y)
      end
      tester:assert(gradcheck(f2to3, {x=torch.randn(3,3)}), "Incorrect gradient")

      local function f3to4(params)
         local y = torch.repeatTensor(params.x, 2, 2, 2, 2)*3
         return torch.sum(y)
      end
      tester:assert(gradcheck(f3to4, {x=torch.randn(3,3,3)}), "Incorrect gradient")


      local function f4to5(params)
         local y = torch.repeatTensor(params.x, 2, 2, 2, 2)*3
         return torch.sum(y)
      end

      tester:assert(autograd(f4to5)({x=torch.randn(3,3,3):cuda()}))


      -- tester:assert(gradcheck(f3to4, {x=torch.randn(3,3,3):float()}), "Incorrect gradient")
   end,
```

**Comment by alexbw** [comment]
Fixed via e436d1d
